### PR TITLE
[11.0][resource_calendar_schedule_iterations]

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
When you want to plan a number of days in the past or in the future consiering
a working calendar, Odoo limits the number of days/hours that you can plan
ahead.

In case that you want to plan days forward/backward, it currently restricts
to 100 iterations.

In case that you want to plan hours forward/backward, it currently restricts
to 100 iterations.

This module allows you to increase the iteration limit used in the resource
calendar to schedule days or hours by means of a system parameter defined
by the administrator.

Configuration
=============

#. Go to *Ŝettings* and activate the developer mode.

#. Go to *Settings / Technical / Parameters / System Parameters* and define,
   new values for the parameters *resource.calendar.schedule.days.iteration.limit*
   and/or *resource.calendar.schedule.hours.iteration.limit*

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please avoid plurals on module name.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by LoisRForgeFlow** [comment]
Just added test to this, hoping for a merge when :green_apple: (:crossed_fingers: )

File: resource_calendar_schedule_iteration/hooks.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            # avoid infinite loops
+            iterations += 1
+
+        return intervals
+
+    if not hasattr(ResourceCalendar, '_schedule_days_original'):
+        ResourceCalendar._schedule_days_original = \
+            ResourceCalendar._schedule_days
+
+    ResourceCalendar._schedule_days = _new_schedule_days
```
Comment (lmignon) [inline resource_calendar_schedule_iteration/hooks.py]: @jbeficent you should use `BaseModel._patch_method("_schedule_days", _new_schedule_days)` https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4281

Comment (lmignon) [inline resource_calendar_schedule_iteration/hooks.py]: @jbeficent you should use `ResourceCalendar._patch_method("_schedule_days", _new_schedule_days)` https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4281

File: resource_calendar_schedule_iteration/hooks.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import pytz
+import datetime
+
+from datetime import timedelta
+
+from odoo.addons.resource.models.resource import ResourceCalendar
+from odoo.tools.float_utils import float_compare
+
+
+def to_naive_user_tz(datetime, record):
```
Comment (lmignon) [inline resource_calendar_schedule_iteration/hooks.py]: @jbeficent Why do you redefine this method already defined into `odoo.addons.resource.models.resource` ?

Why not
`from odoo.addons.resource.models.resource import to_naive_user_tz`

**Review by lmignon — CHANGES_REQUESTED** [review]
@jbeficent a little comment otherwise LGTM. 
Do you plan to propose a change on Odoo master to make this limit configurable by default?

File: resource_calendar_schedule_iteration/hooks.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply resource_calendar_schedule_iteration/hooks.py]: Don.e

File: resource_calendar_schedule_iteration/hooks.py (hunk 2 above)
Comment (LoisRForgeFlow) [reply resource_calendar_schedule_iteration/hooks.py]: Agree

**Comment by LoisRForgeFlow** [comment]
@lmignon Added your suggestions. Maybe it can be proposed for v12 or master, which one do you think has the more chances to get accepted?

**Comment by lmignon** [comment]
> @lmignon Added your suggestions. Maybe it can be proposed for v12 or master, which one do you think has the more chances to get accepted?

@lreficent IMO only master

**Review by lmignon — APPROVED** [review]

File: resource_calendar_schedule_iteration/README.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/server-tools-11-0/server-tools-11-0-resource_calendar_schedule_iteration
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/149/11.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+When you want to plan a number of days in the past or in the future consiering
```
Comment (pedrobaeza) [inline resource_calendar_schedule_iteration/README.rst]: s/consiering/considering

File: resource_calendar_schedule_iteration/README.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+a working calendar, Odoo limits the number of days/hours that you can plan
+ahead.
+
+In case that you want to plan days forward/backward, it currently restricts
+to 100 iterations.
+
+In case that you want to plan hours forward/backward, it currently restricts
```
Comment (pedrobaeza) [inline resource_calendar_schedule_iteration/README.rst]: Repeated sentence

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza Attended your comments.

@lmignon Resources were refactored in v12, so I will first migrate this to v12 when this PR gets merged and then propose a current fix to master.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**